src-tests/test-rpc.lisp

```diff
@@ -7,87 +7,106 @@
 
 (defparameter *expected-response* "test-response")
 
-(defun test-method (&key (sleep 1 sleep-p))
-  (when sleep-p
+(defun test-method (&key (sleep 1 slee-p))
+  (when slee-p
     (sleep sleep))
   "test-response")
 
-
 (deftest test-client-server-dialogue ()
-  (let* ((server-function
-           (lambda ()
-             (let ((dt (rpcq:make-dispatch-table)))
-               (rpcq:dispatch-table-add-handler dt 'test-method)
-               (rpcq:start-server :dispatch-table dt
-                                  :listen-addresses '("inproc://RPCQ-test")))))
-         (server-thread (bt:make-thread server-function)))
-    (sleep 1)
-    (unwind-protect
-      ;; hook up the client
-      (rpcq:with-rpc-client (client "inproc://RPCQ-test")
-        ;; send a communique
-        (let ((server-response (rpcq:rpc-call client "test-method")))
-          (is (string= *expected-response* server-response))))
-      ;; kill the server thread
-      (bt:destroy-thread server-thread))))
+  (with-unique-rpc-address (addr)
+    (let* ((server-function
+             (lambda ()
+               (let ((dt (rpcq:make-dispatch-table)))
+                 (rpcq:dispatch-table-add-handler dt 'test-method)
+                 (rpcq:start-server :dispatch-table dt
+                                    :listen-addresses (list addr)))))
+           (server-thread (bt:make-thread server-function)))
+      (sleep 1)
+      (unwind-protect
+           ;; hook up the client
+           (rpcq:with-rpc-client (client addr)
+             ;; send a communique
+             (let ((server-response (rpcq:rpc-call client "test-method")))
+               (is (string= *expected-response* server-response))))
+        ;; kill the server thread
+        #+ccl
```

Comment:
can we document this weirdness here?

Reply:
and maybe factor it out into a utility function?

Reply:
FWIW, my understanding is that this weirdness is "supposed" to go away "soon" with the resolutions of #75 and #61, which is why I didn't object to it being a little ugly for the moment.

Reply:
Agreed with eric